pkg/agent/util/net_windows.go

```diff
@@ -897,60 +897,239 @@ func VirtualAdapterName(name string) string {
+	out, err := ps.RunCommand(cmd)
 	if err != nil {
-		klog.ErrorS(err, "Failed to get address", "iface", iface)
-		return nil, err
+		return "", err
 	}
-	gw, routes, err := GetNetRoutesOnAdapter(iface.Index)
+	out = strings.TrimSpace(out)
+	return out, err
+}
+
+func renameHostInterface(oriName string, newName string) error {
+	cmd := fmt.Sprintf(`Get-NetAdapter -Name "%s" | Rename-NetAdapter -NewName "%s"`, oriName, newName)
+	_, err := ps.RunCommand(cmd)
 	if err != nil {
-		klog.ErrorS(err, "Failed to get routes", "iface.Index", iface.Index)
-		return nil, err
+		klog.InfoS("Failed to rename adapter", "oriName", oriName, "error", err.Error())
 	}
-	return &config.AdapterNetConfig{
-		Name:    uplinkIfName,
-		Index:   iface.Index,
-		MAC:     iface.HardwareAddr,
-		IPs:     addrs,
-		Routes:  routes,
-		Gateway: gw,
-		MTU:     iface.MTU,
-	}, nil
+	return err
 }
 
-// TODO: Implement MoveIFConfigurations for windows
-func MoveIFConfigurations(ips []*net.IPNet, routes []interface{}, mac net.HardwareAddr, mtu int, from string, to string) error {
-	return nil
+func GetVMSwitchInterfaceName() (string, error) {
+	cmd := fmt.Sprintf(`Get-VMSwitchTeam -Name "%s" | select  NetAdapterInterfaceDescription |  Format-Table -HideTableHeaders`, LocalVMSwitch)
+	out, err := ps.RunCommand(cmd)
+	if err != nil {
+		klog.ErrorS(err, "Failed to find switch with teaming enable.d", "switch", LocalVMSwitch)
+		return "", err
+	}
+	out = strings.TrimSpace(out)
+	// {Amazon Elastic Network Adapter #2} remove flower brackets
+	out = out[1 : len(out)-1]
+	cmd = fmt.Sprintf(`Get-NetAdapter -InterfaceDescription "%s" | select Name | Format-Table -HideTableHeaders`, out)
+	out, err = ps.RunCommand(cmd)
+	if err != nil {
+		klog.ErrorS(err, "Failed to find adapter", "InterfaceDescription", out)
+		return "", err
+	}
+	out = strings.TrimSpace(out)
+	return out, err
 }
 
-// TODO: Implement GetNetRoutesOnAdapter for windows
-func GetNetRoutesOnAdapter(linkIndex int) (string, []interface{}, error) {
-	return "", nil, nil
+func VmSwitchExists() (bool, error) {
+	cmd := fmt.Sprintf(`Get-VMSwitch -Name "%s" -ComputerName $(hostname)`, LocalVMSwitch)
+	_, err := ps.RunCommand(cmd)
+	if err == nil {
+		return true, nil
+	}
+	if strings.Contains(err.Error(), fmt.Sprintf(`unable to find a virtual switch with name "%s"`, LocalVMSwitch)) {
+		return false, nil
+	}
+	return false, err
 }
 
-func RenameInterface(from, to string) error {
-	var renameErr error
-	pollErr := wait.Poll(time.Millisecond*100, time.Second, func() (done bool, err error) {
-		renameErr = renameHostInterface(from, to)
-		if renameErr != nil {
-			klog.InfoS("Unable to rename host interface name with error, retrying", "oldName", from, "newName", to, "err", renameErr)
-			return false, nil
+// This function creates a virtual switch and enables openvswitch extension.
+// If switch exists and extension is enabled, then it will return with no error.
+// Other it will throw an error.
+
+// Case: Multiple interface
```

Comment:
One comment was for supporting multiple interface case. I will add a TODO for it.
The other comments were related to changing the interfacename/ip in the externalnode. 

Lets suppose there is a VM with 2 interfaces. (eth0 and eth1).  Initially the user creates externalnode with eth0, agent boots up and everything is fine. Now if the users update externalnode to have eth1 interface instead of eth0, what would be the desired behavior.